epidose/device/util.sh

```diff
@@ -129,6 +136,54 @@ run_python()
   "$PYTHON" "$SCRIPT_DIR"/"$prog".py $DEBUG_FLAG $VERBOSE_FLAG "$@"
 }
 
+# Check the validity of the cuckoo filter
+# preconditions: none 
+# Returns, through the stdout,
+# the time until a Cuckoo filter is valid (in seconds)
+# If 0 is returned, the Cuckoo filter is stale.
+get_filter_validity_age()
+{
+  if [ -r "$FILTER" ] ; then
+    filter_mtime=$(stat -c '%Y' "$FILTER")
+    time_now=$(date +%s)
+    filter_age=$((time_now - filter_mtime))
+    if [ $filter_age -lt $MAX_FILTER_AGE ] ; then
+      log "Filter's age ($filter_age s) is current; no update required"
+      local validity_time=$((MAX_FILTER_AGE - filter_age))
+      echo "$validity_time"
+    else
+      log "Filter's age ($filter_age s) makes it stale; update required"
+      local validity_time=0
+      echo "$validity_time"
+    fi
+  else
+    log "No filter available; download required"
+    mkdir -p "$(dirname $FILTER)"
+  fi	
+}
+
+# Obtain a (new) version of the Cuckoo filter
+# preconditions: none 
+# Returns, through the stdout, an exit code that defines
+# whether a request to the ha-sever, to fetch a new cuckoo filter,
+# was successful.
+# If 1 is returned, then cuckoo filter was not obtained.
+is_trying_to_get_new_filter()
+{
+  if err=$(curl --silent --show-error --fail --output "$FILTER.new" \
+    "$SERVER_URL/filter" 2>&1) ; then
+    # Atomically replace existing filter with new one
+    mv "$FILTER.new" "$FILTER"
+    log "New filter obtained: $(stat -c %s "$FILTER") bytes"
+    local exit_code=0
+    echo "$exit_code"
+  else
+    log "Unable to get filter: $err"
+    local exit_code=1
+    echo "$exit_code"
```

Comment:
Ditto

Reply:
Ok done.
I will test them and submit a squashed commit.
Regarding the shellcheck, I thought it was running as a pre-commit hook, no?

Reply:
Ok done.

black................................................(no files to check)Skipped
flake8...............................................(no files to check)Skipped
Test shell scripts with shellcheck.......................................Passed